lib/tests/modules/declaration-positions.nix

```diff
@@ -0,0 +1,42 @@
+{ lib, options, ... }:
+let discardPositions = lib.mapAttrs (k: v: v);
+in
+{
```

Comment:
Let's make our test assumptions explicit. Something like

```suggestion
# unsafeGetAttrPos is unspecified best-effort behavior, so we only want to consider this test on an evaluator that satisfies some basic assumptions about this function.
assert builtins.unsafeGetAttrPos "a" { a = true; } != null;
assert builtins.unsafeGetAttrPos "a" (discardPositions { a = true; }) == null;
{
```

(sorry to break the line numbers by adding lines)